tests/integration/relations/test_osm_mysql.py

```diff
@@ -100,22 +100,31 @@ async def test_deploy_and_relate_osm_bundle(ops_test: OpsTest) -> None:
             timeout=1000,
         )
 
+        # osm-zookeeper is never `active` long enough (15 seconds is necessary).
+        # it constantly changes state `active`<>`maintenance`:
+        # > osm-zookeeper/0 [idle] maintenance: Sending Zookeeper configuration
         await ops_test.model.wait_for_idle(
-            apps=[APP_NAME, "osm-kafka", "osm-zookeeper", "osm-mongodb"],
+            apps=[APP_NAME, "osm-kafka", "osm-mongodb"],
             status="active",
             raise_on_blocked=True,
             timeout=1000,
         )
+        await ops_test.model.wait_for_idle(
+            apps=["osm-zookeeper"],
+            status="active",
+            idle_period=1,
+            raise_on_blocked=True,
+            timeout=1000,
+        )
```

Comment:
(please feel free to disregard) another method we could use here is `ops_test.model.block_until(ops_test.model.applications["osm-zookeeper"].status == "active")`. this will unblock the first time the zookeeper charm moves into active status

Reply:
TIL. Improved as suggested.